File: cashflow/src/main/java/com/backend/cashflow/verification/ColumnVerify.java
package com.backend.cashflow.verification;

import com.backend.cashflow.enums.StringLength;
import org.springframework.stereotype.Component;

@Component
public class ColumnVerify {

    public boolean isColumnValid(String s, StringLength field) {
        return s != null &&
                !s.isEmpty() &&
                !s.isBlank() &&
                s.length() > field.getMinLength() && s.length() < field.getMaxLength();
    }

    public boolean isColumnValueValid(double value, StringLength field) {
        return value >= field.getMinLength() && value <= field.getMaxLength();
    }
}

File: cashflow/src/main/java/com/backend/cashflow/verification/ColumnVerify.kt
package com.backend.cashflow.verification

import com.backend.cashflow.enums.StringLength
import org.springframework.stereotype.Component

@Component
class ColumnVerify {
    fun isColumnValid(s: String, field: StringLength): Boolean {
        return s != null &&
                s.isNotEmpty() &&
                s.isNotBlank() &&
                s.length > field.minLength && s.length < field.maxLength;
    }

    fun isColumnValueValid(value: Double, field: StringLength): Boolean {
        return value != null &&
                value >= field.minLength && value <= field.maxLength;
    }
}